Add developer call blog post (#681)
jellyfin/jellyfin.org: blog/2023/10-01-a-call-for-developers/index.mdx

---
title: 'A Call for Developers'
description: "Jellyfin needs your help to get better, so join us this October and beyond!"
authors: joshuaboniface
date: 2023-10-01
slug: a-call-for-developers
tags: [project]
---

![I want YOU to contribute to Jellyfin](./contribute-to-jellyfin.jpg)

This year, DigitalOcean's [Hacktoberfest](https://hacktoberfest.org) is in it's 10th year. In previous years, we haven't made too much of it, aside from adding labels to our subprojects to allow contributors to count their contributions towards rewards. While those are a little less impressive this year, within our project, this is our most important year yet.

We'd like to call on you, the wider community, to help make Jellyfin better! We need contributors, fresh ideas and blood to help the project move past our current funk and into something more.

Will you join us?

<!--truncate-->

> We keep a close eye on the buzz about Jellyfin around Reddit and the Internet more broadly, even if we've closed our own subreddit. The last couple weeks have seen a few major Plex-related kerfufles and the comments inevitably include the following exchange:

Reviewer: I don’t think it’s necessary to mention closing the subreddit 🤷‍♂️

    Person A: Use Jellyfin instead
    Person B: Jellyfin sucks, it lacks client X or feature Y
    Person C: Works for me
    Person D: It needs polish and more apps

Sometimes one of us will jump in, but I think it's high time for us to make a public request for something that's been on our minds as a team for quite a while.

For those who are not aware, Jellyfin is a bit unusual, even within the FLOSS world. We are *explicitly anti-commercial*. Our team is entirely composed of volunteers who contribute because they like what we're doing and can help out in whatever ways they can. This includes developers, translators, and community members who help moderate the community and assist others in troubleshooting and working through problems.

> No one gets paid for Jellyfin work, by design. We're not backed by any company or "organization" with its own agenda. We have no "monetization plan" or anything of the sort, and actively avoid bug bounties. Jellyfin isn't a "product" in the commercial sense of the word. And while we do take donations, these *only* cover our infrastructure costs and the rare piece of developer client hardware as needed; we don't pay developers out of donations, and never will. Mmoney is not an issue (for what we use it for), and that's not really what we need. This is a choice we've made from day 1 to avoid Jellyfin following in the footsteps of nearly every other player in this space, who start out as FLOSS, start adding user-hostile features in the name of money, and then go closed source. We want to be different. Want to be volunteer-only and working on this thing to help make media streaming available to everyone.

Reviewer 2: ```suggestion
No one gets paid for Jellyfin work, by design. We're not backed by any company or "organization" with its own agenda. We have no "monetization plan" or anything of the sort, and actively avoid bug bounties. Jellyfin isn't a "product" in the commercial sense of the word. And while we do take donations, these *only* cover our infrastructure costs and the rare piece of developer client hardware as needed; we don't pay developers out of donations, and never will. Money is not an issue (for what we use it for), and that's not really what we need. This is a choice we've made from day 1 to avoid Jellyfin following in the footsteps of nearly every other player in this space, who start out as FLOSS, start adding user-hostile features in the name of money, and then go closed source. We want to be different. Want to be volunteer-only and working on this thing to help make media streaming available to everyone.
```

But, the team behind Jellyfin is under strain. We love what we do, and want to keep doing it. But our core contributor base is actually fairly small, at most about 30 active people, for everything - the core server, the webUI, and all of our clients. And it hasn't really grown in quite some time. Some of our developers have been with us since the early days, some joined later, but life sometimes gets in the way, and we've lost some over the years, so the number has been pretty flat for the last 3 years all together. For some parts of the project, especially individual apps, we have at most 1 or 2 people working on them, for free in their own time, because it "scratches their itch".

> Some very-well-request features/clients have gotten *no* traction at all, with no one coming in wanting to develop them. We've had to abandon some (like Chromecast for a while, though it's getting new life in the last few weeks) because of this.

Reviewer 3: ```suggestion
Some very-well-requested features/clients have gotten *no* traction at all, with no one coming in willing to start/help developing them. We've had to abandon some (like Chromecast for a while, though it's getting new life in the last few weeks) because of this.
```

> We're aware that probably the biggest complaints about Jellyfin are the lack of client support, and the rough edges/lack of polish. We do hear you. We do want this to improve.

Reviewer 3: ```suggestion
We're aware that probably the biggest complaints about Jellyfin are about the lack of client support, and the rough edges/lack of polish. We do hear you. We do want this to improve this as much as you do.
```

> But we need people to help do so. We need more volunteers who can help make the code better, write new code, and improve things. We need your help to push past what I call the [Development Bystander Problem](https://www.boniface.me/problems-in-floss-3/), get some new blood into the project, and especially, help make it better!

Reviewer 3: ```suggestion
But we need people to help us do so. We need more volunteers who can help making the code better, write new code, and improve things. We need **your** help to push past what I call the [Development Bystander Problem](https://www.boniface.me/problems-in-floss-3/), get some new blood into the project, and especially, helping to make it better!
```

Author: For this one "making" feels incorrect, but I'll accept it.

So, this is our open call to the world: Jellyfin needs YOU!

> Whether you're a developer, a writer, a UI/UX designer, a non-English-language speaker, or just avid enthusiast willing to help out, no matter your experience level, we'd love the extra help.

Reviewer: ```suggestion
Whether you're a developer, a writer, a UI/UX designer, a non-English-language speaker, or just an avid enthusiast willing to help out, no matter your experience level, we'd love the extra help.
```

> Whether you're a developer, a writer, a non-English-language speaker, or just avid enthusiast willing to help out, no matter your experience level, we'd love the extra help.

Reviewer: Would be good to call out ui/ux people here also.

> We have [some resources](https://jellyfin.org/contribute) to help you get started, and our [chat channels on Matrix/IRC/Discord](https://jellyfin.org/contact) and [Forum](https://forum.jellyfin.org) are quite active, with the team very happy to help new contributors.

Reviewer 3: ```suggestion
We have [some resources](https://jellyfin.org/contribute) to help you getting started, and our [chat channels on Matrix/IRC/Discord](https://jellyfin.org/contact) and the [Forum](https://forum.jellyfin.org) are quite active, with the team very happy to help new contributors.
```

Author: Another grammar one, "get started" is definitely correct unless you add "in" before it. But "the" was indeed missing.

> Every little bit helps, from fixing a small bug to working on a full client. We do have [a "track", so to speak](https://github.com/jellyfin/jellyfin-meta/blob/master/policies-and-procedures/new-team-member-nominations.md), for people to officially join the team as well, so if this is something you want, it's absolutely possible after proving yourself with quality contributions and a willingness to participate.

Reviewer 3: ```suggestion
Every little bit helps, from fixing a small bug to improving the documentation or working on a full client. We do have [a "track", so to speak](https://github.com/jellyfin/jellyfin-meta/blob/master/policies-and-procedures/new-team-member-nominations.md), for people to officially join the team as well - so if this is something you want, it's absolutely possible after proving yourself with quality contributions of any kind and a willingness to participate.
```
